src/components/Ques.tsx

```diff
@@ -0,0 +1,173 @@
+import { useRef, useEffect, useState } from "react";
+import styled from "styled-components";
+
+const QuesTitle = styled.h2`
+  color: #07889b;
+  margin-bottom: 20px;
+  text-align: center;
+  font-size: 1.5rem;
+`;
+
+const FormContainer = styled.div`
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  padding: 50px;
+  background-color: #f5f5f5;
+  border-radius: 10px;
+  box-shadow: 0 4px 8px rgba(0, 0, 0, 0.1);
+  max-width: 600px;
+  margin: auto;
+  width: 100%;
+`;
+
+const FormGroup = styled.div<{ isRange: boolean }>`
+  display: flex;
+  flex-direction: ${({ isRange }) => (isRange ? "column" : "row")};
+  align-items: center;
+  background-color: #e0e0e0;
+  padding: 10px 20px;
+  border-radius: 20px;
+  margin-bottom: 15px;
+  width: 100%;
+
+  &:hover {
+    background-color: #f0f0f0;
+  }
+
+  label {
+    margin-left: 10px;
+    font-size: 1rem;
+    cursor: pointer;
+    flex-grow: 1;
+  }
+`;
+
+const RadioInput = styled.input`
+  margin-right: 10px;
+`;
+
+const CheckboxInput = styled.input`
+  margin-right: 10px;
+`;
+
+const DropdownSelect = styled.select`
+  color: #222831;
+  background-color: #e0e0e0;
+  border-color: f5f5f5;
+  width: 100%;
+  padding: 10px;
+  border-radius: 20px;
+  border: 1px solid #ccc;
+`;
+
+const RangeInput = styled.input`
+  width: 100%;
+  margin: 10px 0;
+`;
+
+const RangeLabels = styled.div`
+  display: flex;
+  justify-content: space-between;
+  width: 100%;
+`;
+
+enum Type {
+  RADIO = "radio",
+  CHECKBOX = "checkbox",
+  RANGE = "range",
+  DROPDOWN = "dropdown",
+}
+
+interface Option {
+  optionId: number;
+  optionLabel: string;
+}
+
+interface QuesProps {
+  id: number;
+  title: string;
+  type: Type;
+  options: Option[];
+}
+
+export function Ques(props: QuesProps) {
+  const { id, title, type, options } = props;
+  const formRef = useRef<HTMLFormElement>(null);
+  const [rangeValue, setRangeValue] = useState(0);
+
+  useEffect(() => {
+    if (formRef.current) {
+      formRef.current.reset();
+    }
+  }, [id]);
+
+  const handleRangeChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setRangeValue(Number(event.target.value));
+  };
+
+  return (
```

Comment:
this return statement would look cleaner if you were to use a switch case here. For example:

```
switch(type) {

  case Type.RADIO:
      return your code for radio buttons

   case Type.CHECKBOX:
        return your checkbox code
```